src/main/java/org/jenkinsci/lib/configprovider/ConfigProvider.java

```diff
@@ -128,4 +129,46 @@ public boolean supportsFolder() {
         return true;
     }
 
+
+    /**
+     * @deprecated Use <tt>GlobalConfigFiles.get().getConfigs()</tt> instead.
+     */
+    @Deprecated
+    public Collection<Config> getAllConfigs() {
```

Comment:
although this makes the API some how compatible again, it is semantically wrong. `GlobalConfigFiles.get().getConfigs();` just returns the global configuration files and none in any folders. So I don't think this is a good idea. 

The same applies to all the other methods...